revoke_handler.go

```diff
@@ -0,0 +1,109 @@
+package fosite
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// Implements
+// * https://tools.ietf.org/html/rfc7009#section-2.1
+// The authorization server first validates the client credentials (in
+// case of a confidential client) and then verifies whether the token
+// was issued to the client making the revocation request.  If this
+// validation fails, the request is refused and the client is informed
+// of the error by the authorization server as described below.
+//
+// In the next step, the authorization server invalidates the token.
+// The invalidation takes place immediately, and the token cannot be
+// used again after the revocation.
+//
+// * https://tools.ietf.org/html/rfc7009#section-2.2
+// An invalid token type hint value is ignored by the authorization
+// server and does not influence the revocation response.
+func (f *Fosite) NewRevocationRequest(ctx context.Context, r *http.Request) error {
+	if r.Method != "POST" {
+		return errors.Wrap(ErrInvalidRequest, "HTTP method is not POST")
+	} else if err := r.ParseForm(); err != nil {
+		return errors.Wrap(ErrInvalidRequest, err.Error())
+	}
+
+	clientID, clientSecret, ok := r.BasicAuth()
+	if !ok {
+		return errors.Wrap(ErrInvalidRequest, "HTTP Authorization header missing or invalid")
+	}
+
+	client, err := f.Store.GetClient(clientID)
+	if err != nil {
+		return errors.Wrap(ErrInvalidClient, err.Error())
+	}
+
+	// Enforce client authentication
+	if err := f.Hasher.Compare(client.GetHashedSecret(), []byte(clientSecret)); err != nil {
+		return errors.Wrap(ErrInvalidClient, err.Error())
+	}
+
+	r.ParseForm()
+	token := r.PostForm.Get("token")
+
+	var tokenType = TokenType(r.PostForm.Get("token_type_hint"))
+
+	switch tokenType {
+	case AccessToken:
+	case RefreshToken:
+	default:
+		tokenType = RefreshToken
```

Comment:
See comment above, I think that we should leave tokenType empty if it is not given.